### PR TITLE
Boot load repos in api

### PR DESCRIPTION
**Pull request by rebstar6**
🍾

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/route/VinylDNSService.scala [hunk 1]
```diff
@@ -117,6 +121,12 @@ class VinylDNSService(
     with VinylDNSJsonProtocol
     with JsonValidationRejection {
 
+  val aws4Authenticator = new Aws4Authenticator
+  val authPrincipalProvider =
+    new MembershipAuthPrincipalProvider(userRepository, membershipRepository)
```
Comment (rebstar6) [inline modules/api/src/main/scala/vinyldns/api/route/VinylDNSService.scala]: those formerly lazy vals are created here ^

File: modules/core/src/main/scala/vinyldns/core/repository/DataStoreLoader.scala [hunk 2]
```diff
@@ -20,27 +20,33 @@ import cats.data._
...
 
 object DataStoreLoader {
+
+  private val logger = LoggerFactory.getLogger("DataStoreLoader")
+
   def loadAll[A <: DataAccessor](
       configs: List[DataStoreConfig],
       crypto: CryptoAlgebra,
-      dataAccessorProvider: DataAccessorProvider[A]): IO[DataAccessor] =
+      dataAccessorProvider: DataAccessorProvider[A]): IO[A] =
```
Comment (rebstar6) [inline modules/core/src/main/scala/vinyldns/core/repository/DataStoreLoader.scala]: oops

File: modules/core/src/main/scala/vinyldns/core/repository/DataStoreLoader.scala [hunk 3]
```diff
@@ -20,27 +20,33 @@ import cats.data._
...
     for {
       activeConfigs <- IO.fromEither(getValidatedConfigs(configs, dataAccessorProvider.repoNames))
       dataStores <- activeConfigs.map(load(_, crypto)).parSequence
       accessor <- IO.fromEither(generateAccessor(dataStores, dataAccessorProvider))
     } yield accessor
 
-  def load(config: DataStoreConfig, crypto: CryptoAlgebra): IO[(DataStoreConfig, DataStore)] =
+  def load(config: DataStoreConfig, crypto: CryptoAlgebra): IO[(DataStoreConfig, DataStore)] = {
+    logger.error(s"Attempting to load repos ${config.repositories.keys} from ${config.className}")
```
Comment (rebstar6) [inline modules/core/src/main/scala/vinyldns/core/repository/DataStoreLoader.scala]: added a few logs for good measure. we should only see these messages once on startup

File: modules/dynamodb/src/it/scala/vinyldns/dynamodb/repository/DynamoDBDataStoreProviderIntegrationSpec.scala [hunk 4]
```diff
@@ -77,10 +78,10 @@ class DynamoDBDataStoreProviderIntegrationSpec extends DynamoDBIntegrationSpec {
       )
       val userRepo = dataStore.get[UserRepository](user)
 
-      val save = userRepo.map(_.save(testUser)).parSequence
+      val save = userRepo.map(_.save(testUser)).sequence[IO, User]
       save.unsafeRunSync() shouldBe Some(testUser)
 
-      val get = userRepo.map(_.getUser(testUser.id)).parSequence
+      val get = userRepo.map(_.getUser(testUser.id)).sequence[IO, Option[User]]
```
Comment (rebstar6) [inline modules/dynamodb/src/it/scala/vinyldns/dynamodb/repository/DynamoDBDataStoreProviderIntegrationSpec.scala]: my b, should have used sequence in the 1st place

File: modules/api/src/main/scala/vinyldns/api/VinylDNSConfig.scala [hunk 5]
```diff
@@ -32,22 +32,12 @@ object VinylDNSConfig {
...
-  lazy val recordSetStoreConfig: DynamoDBRepositorySettings = DynamoConfig.recordSetStoreConfig
-  lazy val recordChangeStoreConfig: DynamoDBRepositorySettings =
-    DynamoConfig.recordChangeStoreConfig
-  lazy val usersStoreConfig: DynamoDBRepositorySettings = DynamoConfig.usersStoreConfig
-  lazy val groupsStoreConfig: DynamoDBRepositorySettings = DynamoConfig.groupsStoreConfig
-  lazy val groupChangesStoreConfig: DynamoDBRepositorySettings =
-    DynamoConfig.groupChangesStoreConfig
-  lazy val membershipStoreConfig: DynamoDBRepositorySettings = DynamoConfig.membershipStoreConfig
-
+  lazy val dataStoreConfigs: List[DataStoreConfig] =
```
[the remark this answers is not in the document]
Comment (rebstar6) [reply modules/api/src/main/scala/vinyldns/api/VinylDNSConfig.scala]: are we set on fully retiring that? im gonna leave in VinylDNSConfig, but load it in IO - update pending

File: modules/api/src/test/scala/vinyldns/api/route/HealthCheckRoutingSpec.scala [hunk 6]
```diff
@@ -41,6 +41,8 @@ class HealthCheckRoutingSpec
 
   private val mockZoneRepo = mock[ZoneRepository]
   val healthService = new HealthService(mockZoneRepo)
+  val vinylDNSAuthenticator: VinylDNSAuthenticator = new TestVinylDNSAuthenticator(
```
[the remark this answers is not in the document]
Comment (rebstar6) [reply modules/api/src/test/scala/vinyldns/api/route/HealthCheckRoutingSpec.scala]: it extends VinylDNSDirectives, which it should not have.

removing

File: modules/api/src/universal/conf/application.conf [hunk 7]
```diff
@@ -135,6 +120,7 @@ vinyldns {
   # the max number of changes in a single batch change.  Change carefully as this has performance
   # implications
   batch-change-limit = 20
+
```
[the remark this answers is not in the document]
Comment (rebstar6) [reply modules/api/src/universal/conf/application.conf]: maybe...

File: modules/dynamodb/src/main/scala/vinyldns/dynamodb/repository/DynamoDBMembershipRepository.scala [hunk 8]
```diff
@@ -32,17 +32,19 @@ object DynamoDBMembershipRepository {
...
   def apply(
       config: DynamoDBRepositorySettings,
       dynamoConfig: DynamoDBDataStoreSettings): IO[DynamoDBMembershipRepository] = {
 
-    val dynamoDBHelper = new DynamoDBHelper(
-      DynamoDBClient(dynamoConfig),
-      LoggerFactory.getLogger("DynamoDBMembershipRepository"))
+    val tableName = config.tableName
+
+    log.error(s"Initializing DynamoDBMembershipRepository with name $tableName")
```
[the remark this answers is not in the document]
Comment (rebstar6) [reply modules/dynamodb/src/main/scala/vinyldns/dynamodb/repository/DynamoDBMembershipRepository.scala]: were supposedly doing these all in parallel, so im not 100% on what will happen if theyre just in the caller

File: modules/dynamodb/src/main/scala/vinyldns/dynamodb/repository/DynamoDBDataStoreProvider.scala [hunk 9]
```diff
@@ -74,10 +74,14 @@ class DynamoDBDataStoreProvider extends DataStoreProvider {
...
-      repoSettings.get(repoName).map(fn(_)).parSequence
-    }
+        fn: DynamoDBRepositorySettings => IO[T]): IO[Option[T]] =
+      for {
+        configuredOn <- IO.pure(repoSettings.get(repoName))
+        _ <- IO(
+          configuredOn.foreach(_ => logger.error(s"Loading dynamodb repo for type: $repoName")))
+        repo <- configuredOn.map(fn(_)).sequence[IO, T]
+        _ <- IO(repo.foreach(_ => logger.error(s"Completed dynamodb load for type: $repoName")))
+      } yield repo
```
Comment (pauljamescleary) [inline modules/dynamodb/src/main/scala/vinyldns/dynamodb/repository/DynamoDBDataStoreProvider.scala]: slight optimization, can eliminate the `foreach` here...

```scala
          repoSettings.get(repoName).map { settings =>
            for {
              _ <- IO(logger.error(s"Loading dynamodb repo for type: $repoName"))
              repo <- fn(settings)
              _ <- IO(logger.error(s"Completed dynamodb load for type: $repoName"))
            } yield repo
          }.sequence
```

Comment (rebstar6) [reply modules/dynamodb/src/main/scala/vinyldns/dynamodb/repository/DynamoDBDataStoreProvider.scala]: yeesh a map to a for. i can do it, but i dunno that it looks much better lol

**Pull request merged**